dbeam-core/src/main/java/com/spotify/dbeam/args/DbeamQueryBuilder.java

```diff
@@ -0,0 +1,184 @@
+    return new DbeamQueryBuilder(String.format("SELECT * FROM %s WHERE 1=1", tableName));
+  }
+
+  public static DbeamQueryBuilder fromSqlQuery(final String sqlQuery) {
+    String s = removeTrailingSymbols(sqlQuery);
+    return new DbeamQueryBuilder(s);
+  }
+
+  public DbeamQueryBuilder withPartitionCondition(
+          String partitionColumn, String startPointIncl, String endPointExcl) {
+    sqlBuilder.append(createSqlPartitionCondition(partitionColumn, startPointIncl, endPointExcl));
+    return this;
+  }
+          
+  // TODO It is assumed now that partitionColumn is of non-numeric type 
+  private static String createSqlPartitionCondition(
+      String partitionColumn, String startPointIncl, String endPointExcl) {
+    return String.format(
+        " AND %s >= '%s' AND %s < '%s'",
+        partitionColumn, startPointIncl, partitionColumn, endPointExcl);
+  }
+
+  public DbeamQueryBuilder withParallelizationCondition(
+      String partitionColumn, long startPointIncl, long endPoint, boolean isEndPointExcl) {
+    sqlBuilder.append(
+        createSqlSplitCondition(partitionColumn, startPointIncl, endPoint, isEndPointExcl));
+    return this;
+  }
+
+  private static String createSqlSplitCondition(
+          String partitionColumn, long startPointIncl, long endPoint, boolean isEndPointExcl) {
+    
+    String upperBoundOperation = isEndPointExcl ? "<" : "<=";
+    return String.format(
+            " AND %s >= %s AND %s %s %s",
+            partitionColumn, startPointIncl, partitionColumn, upperBoundOperation, endPoint);
+  }
+
+  /**
+   * Returns generated SQL query string.
+   * 
+   * @return generated SQL query string.
+   */
+  public String build() {
+    limitStr.map(x -> sqlBuilder.append(x));
+    limitStr = Optional.empty();
+    return sqlBuilder.toString();
+  }
+
+  private static String removeTrailingSymbols(String sqlQuery) {
+    return sqlQuery.replaceAll("[\\s|;]+$", "");
+  }
+
+  public DbeamQueryBuilder withLimit(Optional<Integer> limitOpt) {
+    return limitOpt.map(l -> this.withLimit(l)).orElse(this);
+  }
+  
+  public DbeamQueryBuilder withLimit(long limit) {
+    limitStr = Optional.of(String.format(" LIMIT %d", limit));
+    return this;
+  }
+
+  public DbeamQueryBuilder withLimitOne() {
+    return withLimit(1L);
+  }
+
+  @Override
+  public String toString() {
+    return build();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (obj instanceof DbeamQueryBuilder) {
+      DbeamQueryBuilder that = (DbeamQueryBuilder) obj;
+      return build().equals((that.build())) && limitStr.equals(that.limitStr);
+    }
+    return false;
+  }
+
+  @Override
+  public int hashCode() {
+    return sqlBuilder.hashCode();
+  }
+
+  /**
+   * Generates a new query to get MIN/MAX values for splitColumn.  
+   * 
+   * @param splitColumn column to use
+   * @param minSplitColumnName MIN() column value alias
```

Comment:
I have a feeling we can do away with passing `minSplitColumnName` / `maxSplitColumnName` and just have it fixed here. I think there should be only one caller for this.

Reply:
We use them later in the code.